graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/ClusterAdapterOS2.java

```diff
@@ -292,10 +290,45 @@ private Optional<ClusterHealthResponse> clusterHealth(Collection<String> indices
         }
     }
 
-    private boolean indicesExist(String... indices) {
-        final GetIndexRequest getIndexRequest = new GetIndexRequest(indices);
-        return client.execute((c, requestOptions) -> c.indices().exists(getIndexRequest, requestOptions));
-    }
+    @Override
+    public Optional<HealthStatus> deflectorHealth(Collection<String> indices) {
+        if (indices.isEmpty()) {
+            return Optional.of(HealthStatus.Green);
+        }
 
+        final Map<String, String> aliasMapping = catApi.aliases();
+        final Set<String> mappedIndices = indices
+                .stream()
+                .map(index -> aliasMapping.getOrDefault(index, index))
+                .collect(Collectors.toSet());
 
+        final Set<IndexSummaryResponse> indexSummaries = catApi.indices()
+                .stream()
+                .filter(indexSummary -> mappedIndices.contains(indexSummary.index()))
+                .collect(Collectors.toSet());
+
+        if (indexSummaries.size() < mappedIndices.size()) {
+            return Optional.empty();
+        }
+
+        return indexSummaries.stream()
+                .map(IndexSummaryResponse::health)
+                .map(HealthStatus::fromString)
+                .min((status1, status2) -> {
+                    if (status1.equals(HealthStatus.Red)) {
+                        return status2.equals(HealthStatus.Red) ? 0 : -1;
+                    }
+                    if (status1.equals(HealthStatus.Green)) {
+                        return status2.equals(HealthStatus.Green) ? 0 : 1;
+                    }
+                    if (status2.equals(HealthStatus.Green)) {
+                        return -1;
+                    }
+                    if (status2.equals(HealthStatus.Red)) {
+                        return 1;
+                    }
```

Comment:
Intuitively, I would have thought that `compareTo()` would work, too?

Reply:
to clarify, because the extract seems to small (I thought I selected the full comparison): just compare the enum in min(left, right) - because the ordering inside the enum is, what we expect